nixos/modules/config/stevenblack.nix

```diff
@@ -1,17 +1,17 @@
 { config, lib, pkgs, ... }:
 
 let
-  inherit (lib) optionals mkOption mkEnableOption types mkIf elem concatStringsSep maintainers;
+  inherit (lib) optionals mkOption mkEnableOption types mkIf elem maintainers;
   cfg = config.networking.stevenblack;
 
-  # needs to be in a specific order
-  activatedHosts = with cfg; [ ]
-    ++ optionals (elem "fakenews" block) [ "fakenews" ]
-    ++ optionals (elem "gambling" block) [ "gambling" ]
-    ++ optionals (elem "porn" block) [ "porn" ]
-    ++ optionals (elem "social" block) [ "social" ];
+  pkg = pkgs.stevenblack-blocklist;
 
-  hostsPath = "${pkgs.stevenblack-blocklist}/alternates/" + concatStringsSep "-" activatedHosts + "/hosts";
+  # needs to be in a specific order
+  activatedHosts = with cfg; [ pkg.ads ]
+    ++ optionals (elem "fakenews" block) [ pkg.fakenews ]
+    ++ optionals (elem "gambling" block) [ pkg.gambling ]
+    ++ optionals (elem "porn" block) [ pkg.porn ]
+    ++ optionals (elem "social" block) [ pkg.social ];
```

Comment:
```suggestion
  activatedHosts = [ pkg.ads ]
    ++ optional (elem "fakenews" cfg.block) pkg.fakenews
    ++ optional (elem "gambling" cfg.block) pkg.gambling
    ++ optional (elem "porn" cfg.block) pkg.porn
    ++ optional (elem "social" cfg.block) pkg.social;
```

Reply:
This part is changing because order no longer matters (leftover remnant from old configuration).

This is essentially the new change (largely courtesy of @Gerg-L):

```nix
options.networking.stevenblack = {
  # other options same
  package = mkPackageOption pkgs "stevenblack-blocklist" {};
};
networking.hostFiles = map (lib.getOutput x cfg.package) (cfg.package.ads ++ cfg.block)
```

Reply:
Generally in Nixpkgs, people (not so much me) have been moving away from `lib.optional`, because it's apparently confusing to newcomers and easy to make mistakes with. (If you accidentally do `lib.optional cond [` in `buildInputs` you end up with a nested array which is omitted from `buildInputs`.)

Reply:
Resolved in latest push